tools/ClassGen/InstrBuilder.cpp

```diff
@@ -79,11 +84,23 @@ void InstrBuilder::emitIRBuilderMethods(std::ostream &osH,
     osH << ", " << getStorageTypename(&op.first) << " " << op.second;
     osB << ", " << getStorageTypename(&op.first) << " " << op.second;
   }
-
   osH << ");\n";
+  osB << ") {\n";
+
+  // Create allocations for the scratch operands.
+  for (const auto &op : operands_) {
+    if (op.second == OperandKind::Scratch) {
+      std::string allocSuffix = llvm::StringRef(op.first).lower();
+      osB << "  std::string " << op.first << "Name = name.str() + \"."
+          << allocSuffix << "\";\n";
+      osB << "  auto *" << op.first << "Type = F_->getGraph()->getParent()"
+          << "->uniqueType(ElemKind::Int8QTy, {1}, 0.0, 0);\n";
```

Comment:
Int8QTy is used only for the purpose of reserving 1 byte of memory, but the actual scratch may get a different element type and shape at runtime, right?

Reply:
Also, would we still need this if the memory allocator would properly support 0 sized buffers?

Reply:
I allocate scratch as a 1D vector with the element type always `Int8QTy` and the length returned by the function `getScratchSize()` which is to be interpreted as the size in bytes.
In my vision the scratch is a raw buffer (typeless) required by a given implementation to write some intermediate data and it could contain any data in principle (float, strings, structures, etc). I chose `Int8QTy` as the single type available which has the element size 1 byte although I would have chosen `Int8ITy` if it would have been available in order to not use some dummy scale/offset for this.
So at run-time in principle the buffer could contain anything (the implementation is free to store anything).
I would have made the initial allocation using 0 bytes if the memory allocator would have allowed.